containers/nestjs/src/game/APong.ts

```diff
@@ -200,29 +222,43 @@ class Ball extends Rect {
   }
 
   // TODO: Maybe don't pass this method the players?
-  collide(leftPlayer: Player, rightPlayer: Player) {
-    this.collideWithPaddle(leftPlayer.paddle);
-    this.collideWithPaddle(rightPlayer.paddle);
-
-    this.collideWithBorder(leftPlayer, rightPlayer);
+  collide(leftPlayer: Player, rightPlayer: Player): Sides {
+    const paddleColide: Sides = this.collideWithPaddle(leftPlayer.paddle)
+      ? Sides.LeftPaddle
+      : this.collideWithPaddle(rightPlayer.paddle)
+        ? Sides.RightPaddle
+        : Sides.None;
+
+    const borderCollide: Sides = this.collideWithBorder(
+      leftPlayer,
+      rightPlayer,
+    );
+    return paddleColide != Sides.None ? paddleColide : borderCollide;
   }
 
-  collideWithBorder(leftPlayer: Player, rightPlayer: Player) {
-    if (this.top <= 0 || this.bottom >= WINDOW_HEIGHT) {
-      this._pos.y = this.top < 0 ? 0 : WINDOW_HEIGHT - this._size.h;
-      this._vel.invertdY();
-    }
+  collideWithBorder(leftPlayer: Player, rightPlayer: Player): Sides {
     if (this.left <= 0 || this.right >= WINDOW_WIDTH) {
+      // TODO: Make return value retrieval better
```

Comment:
Idk if this is what you meant by this comment, but you can replace the code below it with a ternary:
`return this.left <= 0 ? Sides.left : Sides.Right;`

Reply:
I put this comment here because I thought at the time that this was an ugly way of checking if it collided with the left border or the right border because we also checked such a thing in the if-statement right above the comment. Looking back at it now it isn't really that bad and it could be pushed to master like this.